python/mage/tgn/definitions/instances.py

```diff
@@ -0,0 +1,418 @@
+"""
+Instances of TGN
+"""
+from typing import Tuple, Dict
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from mage.tgn.constants import (
+    MessageFunctionType,
+    MessageAggregatorType,
+    MemoryUpdaterType,
+    TGNLayerType,
+)
+from mage.tgn.definitions.layers import (
+    TGNLayerGraphSumEmbedding,
+    TGNLayerGraphAttentionEmbedding,
+)
+from mage.tgn.definitions.tgn import TGN
+
+
+class TGNEdgesSelfSupervised(TGN):
```

Comment:
Check if user can be prevented from constructing this.

Reply:
Discussed it, so I will resolve this.